x/stakeibc/keeper/community_pool_test.go

```diff
@@ -0,0 +1,146 @@
+package keeper_test
+
+import (
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/gogoproto/proto"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+
+	epochtypes "github.com/Stride-Labs/stride/v14/x/epochs/types"
+	icqtypes "github.com/Stride-Labs/stride/v14/x/interchainquery/types"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/keeper"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+func (s *KeeperTestSuite) SetupQueryCommunityPoolBalance(icaAccountType types.ICAAccountType) types.HostZone {
+	// We need to register the transfer channel to initialize the light client state
+	s.CreateTransferChannel(HostChainId)
+
+	// Craete host zone
+	// We must use valid addresses for each ICA since they're serialized for the query request
+	depositAddress := s.TestAccs[0]
+	returnAddress := s.TestAccs[1]
+	hostZone := types.HostZone{
+		ChainId:                        HostChainId,
+		ConnectionId:                   ibctesting.FirstConnectionID,
```

Comment:
make the HostDenom a field and set it to Atom since later in the tests you assume Atom... this would allow you to properly check that it is hostZone.hostDenom instead of hard coding it

Reply:
i don't think we ever reference the host denom field from the struct, so I was leaving it off intentionally. We pass the denom in as a param in the query so I was just using the `Atom` field as input and validation

I also think we should try to use the default global variables when possible (e.g. HostChainId, Atom) so we don't have to pass those around to each function.

Lmk if you disagree with any of that!

Reply:
Hmmm, good point since this covers any denom for the stake or unstake routes.  I agree we should use the testing framwork's defaults but I always worry when I see any hardcoded value like Atom but for this case I am convinced.

Reply:
yeah I think if there was conditional logic based on the denom in the query function (which I actually originally thought there was when I started writing this test), then we should be more explicit. But since it's agnostic to the denom, I think we can just use the default